src/java/org/apache/cassandra/net/OutboundConnection.java

```diff
@@ -439,13 +452,14 @@ private void releaseCapacity(long count, long bytes)
     private void onOverloaded(Message<?> message)
     {
         overloadedCountUpdater.incrementAndGet(this);
-        overloadedBytesUpdater.addAndGet(this, canonicalSize(message));
+        
+        int canonicalSize = canonicalSize(message);
```

Comment:
We were calculating the canonical size twice...so fixed that while I was here.

Reply:
`message` caches its `canonicalSize`..